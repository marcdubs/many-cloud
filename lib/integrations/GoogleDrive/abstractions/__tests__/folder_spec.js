const Folder = require("../folder");

describe("folder", () => {
  describe("constructor", () => {
    const expected_id = "1234",
      expected_type = "folder",
      expected_connection = "dummy_connection";
    let actual;
    beforeEach(() => {
      actual = new Folder(expected_id, expected_connection);
    });

    it("sets the connection", () => {
      expect(actual.connection).toEqual(expected_connection);
    });

    it("sets the type", () => {
      expect(actual.type).toEqual(expected_type);
    });

    it("sets the id", () => {
      expect(actual.id).toEqual(expected_id);
    });
  });

  describe("list_files", () => {
    describe("when operation is successfull", () => {
      const expected_files = [
        {
          id: "1234",
          name: "Folder1",
          mimeType: "application/vnd.google-apps.folder"
        },
        {
          id: "4321",
          name: "File1",
          mimeType: "file"
        }
      ];
      let folder, actual, connection;
      beforeEach(async done => {
        connection = {
          list_all_files: jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
              resolve({ files: expected_files });
            });
          })
        };
        folder = new Folder("1234", connection);
        actual = await folder.list_files();
        done();
      });

      it("calls the list_all_files function", () => {
        expect(connection.list_all_files).toHaveBeenCalled();
      });

      it("returns an array of correct folders and files", () => {
        expect(actual).toMatchSnapshot();
      });
    });

    describe("when operation is unsuccessfull", () => {
      const expected_error = "Google it";
      let folder, connection;
      beforeEach(() => {
        connection = {
          list_all_files: jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
              reject(new Error(expected_error));
            });
          })
        };
        folder = new Folder("1234", connection);
      });

      it("rejects with the error", async () => {
        await expect(
          folder.list_files()
        ).rejects.toThrowErrorMatchingSnapshot();
      });
    });
  });
});
