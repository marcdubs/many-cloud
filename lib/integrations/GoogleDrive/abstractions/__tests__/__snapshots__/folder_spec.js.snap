// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`folder list_files when operation is successfull returns an array of correct folders and files 1`] = `
Array [
  Folder {
    "connection": Object {
      "list_all_files": [MockFunction] {
        "calls": Array [
          Array [
            "1234",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      },
    },
    "id": "1234",
    "name": "Folder1",
    "type": "folder",
  },
  File {
    "connection": Object {
      "list_all_files": [MockFunction] {
        "calls": Array [
          Array [
            "1234",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      },
    },
    "id": "4321",
    "name": "File1",
    "type": "file",
  },
]
`;

exports[`folder list_files when operation is unsuccessfull rejects with the error 1`] = `"Google it"`;
